cmd/release.go

```diff
@@ -240,11 +241,17 @@ func getChangelog(version string, rc io.ReadCloser) (string, string) {
 		s       []string
 		header  string
 		reading bool
+
+		releaseHeaderPattern = "## " + strings.ReplaceAll(version, ".", "\\.") + " / \\d{4}-\\d{2}-\\d{2}"
 	)
 	for (len(s) == 0 || reading) && scanner.Scan() {
 		text := scanner.Text()
 		switch {
-		case strings.HasPrefix(text, "## "+version+" "):
+		case strings.HasPrefix(text, "## "+version):
+			if valid, _ := regexp.MatchString(releaseHeaderPattern, text); !valid {
+				fatal(errors.New("Found invalid release header in changelog for version '" + version + "'.  " +
+					"Expected format '## " + version + " / YYYY-MM-DD'. Found '" + text + "'"))
```

Comment:
what about putting `"Expected format = " + releaseHeaderPattern`? That would be the most explicit possible (:

Reply:
I thought the more explicit `YYYY-MM-DD` would be a bit more clear to the user than the pattern of digits.

Reply:
That's fair, but it's easy to de-synchronise those things (e.g regexp will change but we will forget to improve error message). (: If we can avoid this it would be awesome. Maybe put both?

Reply:
I think you are right about the possible synchronization issues, I already had to fix one mistake related to that, so I'll just put the regex directly into the error message as you suggested.